scene/main/window.cpp

```diff
@@ -388,6 +393,26 @@ Size2i Window::get_size() const {
 	return size;
 }
 
+float Window::_get_scale_to_px() const {
+	if (get_embedder() == nullptr && DisplayServer::get_singleton()->get_screen_coordiantes_unit() == DisplayServer::SCREEN_COORDS_UNIT_DPI_ADJUSTED_PIXEL) {
+		if (window_id != DisplayServer::INVALID_WINDOW_ID) {
+			// Window is visible, use window scale directly.
+			return MAX(1.0, DisplayServer::get_singleton()->window_get_scale(window_id));
+		} else {
+			// Window is hidden, use saved window rect to estimate scale.
+			return MAX(1.0, DisplayServer::get_singleton()->screen_get_scale(DisplayServer::get_singleton()->get_screen_from_rect(Rect2i(position, size))));
+		}
+	} else {
+		return 1.0;
```

Comment:
It doesn't make any sense for embedded windows, since the scaling will be already applied to the native window. The goal of this is automatic scaling with screen DPI, not per-window-scaling, which might be useful as a dedicated feature.